src/redisraft.c

```diff
@@ -673,12 +675,54 @@ static void handleRedisCommand(RedisRaftCtx *rr,
         return;
     }
 
-    /* reset asking state when we go to execute the command */
+    handleRedisCommandAppend(rr, ctx, cmds);
+
+    /* reset asking state when after executing the command */
     if (cmds->asking) {
         setAskingState(rr, ctx, false);
     }
+}
 
-    handleRedisCommandAppend(rr, ctx, cmds);
+static void redirectCommand(RedisRaftCtx *rr, RedisModuleCtx *ctx, Node *leader) {
+
+    RedisModule_Assert(!raft_is_leader(rr->raft));
+
+    /* One anomaly here is that may redirect a client to the leader even for
+     * commands have no keys, which is something Redis Cluster never does. We
+     * still need to consider how this impacts clients which may not expect it.
+     */
+    replyRedirect(ctx, 0, &leader->addr);
+}
+
+static RRStatus handleNonLeaderCommand(RedisRaftCtx *rr, RedisModuleCtx *ctx, Node *leader, RaftRedisCommandArray *cmds) {
+    RedisModule_Assert(!raft_is_leader(rr->raft));
+
+    // reply ASK for ASKING state commands
```

Comment:
I guess this is one of our rare "consistent" coding convention, we use `/* */` for comments.